src/Executables/Game/PacketHandlers/TokenLoginHandler.cs

```diff
@@ -48,13 +48,23 @@ public async Task ExecuteAsync(GamePacketContext<TokenLogin> ctx, CancellationTo
                 ctx.Connection.Close();
                 return;
             }
+            
+            // Prevent cross client token forgery
+            var validSession = await _cacheManager.Get<uint>("account:token:"+token.AccountId);
+            if (validSession != 0 && validSession != ctx.Packet.Key)
+            {
+                _logger.LogWarning("Received invalid auth token, session does not match {TokenSession} != {PacketSession}", validSession, ctx.Packet.Key);
+                ctx.Connection.Close();
+                return;
+            }
 
             // todo verify ip address
 
             _logger.LogDebug("Received valid auth token");
 
             // Remove TTL from token so we can use it for another game core transition
             await _cacheManager.Persist(key);
+            await _cacheManager.Expire("account:token:"+token.AccountId, 3600); // TODO: add listener to drop connection when it expires
```

Comment:
Magic number.

Also do we really want to drop connection or refresh the token?

Reply:
Is there any encouraged way for TTL management in the project? If not I could implement such a way to organize it.

Yeah we do, tokens must be disposable for assuring session security and integrity. Since we're not using any JWT like token that self validates, we need to invalidate the tokens that we store and revalidate them via some kind of signature (in this case value of `account:token:` key) for security and maintainability perspective. The block at lines 51-59 serving this purpose also.